### ISSUE TITLE
Adding new Resources after bootstraping shows Error

### ISSUE THREAD

**Issue by HeinrichPet** [International-Data-Spaces-Association/DataspaceConnectorUI#106]
I have loaded some data via bootsrap into the DSC (route missing). When adding new resources the UI shows "Save resource failed."

Log FF:
```
Error on API call:  
Object { status: 404, statusText: "", headers: {…}, config: {…}, request: {…}, data: {…} }
​
config: Object { url: "https://localhost:8080/api/catalogs/8edfc54f-62ba-45eb-8093-f7cf167f7621/offers", method: "post", data: "[\"086f69f0-905b-4ff8-b5bf-4f09d6722410\"]", … }
​
data: Object { message: "Resource not found." }
​
headers: Object { "x-content-type-options": "nosniff", "x-xss-protection": "1; mode=block", "cache-control": "no-cache, no-store, max-age=0, must-revalidate", … }
​
request: Object { _eventsCount: 6, outputSize: 0, writable: true, … }
​
status: 404
​
statusText: ""
​
<prototype>: Object { … }
errorUtils.js:12
Error on API call:  
Object { status: 404, statusText: "", headers: {…}, config: {…}, request: {…}, data: {…} }
​
config: Object { url: "https://localhost:8080/api/catalogs/8edfc54f-62ba-45eb-8093-f7cf167f7621/offers", method: "post", data: "[\"21c25df8-7887-4ea8-b57b-98c066730cb9\"]", … }
​
data: Object { message: "Resource not found." }
​
headers: Object { "x-content-type-options": "nosniff", "x-xss-protection": "1; mode=block", "cache-control": "no-cache, no-store, max-age=0, must-revalidate", … }
​
request: Object { _eventsCount: 6, outputSize: 0, writable: true, … }
​
status: 404
​
statusText: ""
​
<prototype>: Object { … } 
```

Log UI:
```
[1] Error on POST /api/catalogs/8edfc54f-62ba-45eb-8093-f7cf167f7621/offers {"status":404,"statusText":"","headers":{"x-content-type-options":"nosniff","x-xss-protection":"1; mode=block","cache-control":"no-cache, no-store, max-a
ge=0, must-revalidate","pragma":"no-cache","expires":"0","strict-transport-security":"max-age=31536000 ; includeSubDomains","x-frame-options":"DENY","content-type":"application/json","transfer-encoding":"chunked","date":"Thu, 09
Sep 2021 19:02:13 GMT","connection":"close"},"config":{"url":"https://localhost:8080/api/catalogs/8edfc54f-62ba-45eb-8093-f7cf167f7621/offers","method":"post","data":"[\"2ffee76d-ff1c-4b65-846a-d597e011a5bf\"]","headers":{"Accept
":"application/json, text/plain, */*","Content-Type":"application/json","User-Agent":"axios/0.19.0","Content-Length":40},"auth":{"username":"admin","password":"password"},"transformRequest":[null],"transformResponse":[null],"time
out":0,"xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName": [...]

```
Log DSC:
```
2021-09-09T15:36:00,949 [https-jsse-nio-8080-exec-3] WARN - Resolved [io.dataspaceconnector.common.exception.ResourceNotFoundException: CatalogService: 8edfc54f-62ba-45eb-8093-f7cf167f7621]
2021-09-09T15:36:56,346 [scheduling-1] INFO - Scanning agreements...
2021-09-09T15:36:56,528 [https-jsse-nio-8080-exec-1] WARN - Resolved [io.dataspaceconnector.common.exception.ResourceNotFoundException: CatalogService: 8edfc54f-62ba-45eb-8093-f7cf167f7621]
```

Get api/catalogs:
```
{
  "_embedded": {
    "catalogs": [
      {
        "creationDate": "2021-09-09T15:34:46.128+0200",
        "modificationDate": "2021-09-09T15:34:46.128+0200",
        "title": "IDS Catalog",
        "description": "This catalog is created from an IDS infomodel catalog.",
        "additional": {},
        "_links": {
          "self": {
            "href": "https://localhost:8080/api/catalogs/35c5d4b7-edd4-4f89-bb4f-f7d70da287c5"
          },
          "offers": {
            "href": "https://localhost:8080/api/catalogs/35c5d4b7-edd4-4f89-bb4f-f7d70da287c5/offers{?page,size}",
            "templated": true
          }
        }
      },
      {
        "creationDate": "2021-09-09T15:34:46.260+0200",
        "modificationDate": "2021-09-09T15:34:46.260+0200",
        "title": "IDS Catalog",
        "description": "This catalog is created from an IDS infomodel catalog.",
        "additional": {},
        "_links": {
          "self": {
            "href": "https://localhost:8080/api/catalogs/8dc16f8a-73bd-44ec-a658-efafe6d0c561"
          },
          "offers": {
            "href": "https://localhost:8080/api/catalogs/8dc16f8a-73bd-44ec-a658-efafe6d0c561/offers{?page,size}",
            "templated": true
          }
        }
      }
    ]
  },
  "_links": {
    "self": {
      "href": "https://localhost:8080/api/catalogs?page=0&size=30"
    }
  },
  "page": {
    "size": 30,
    "totalElements": 2,
    "totalPages": 1,
    "number": 0
  }
}
```

**Comment by BastianWel** [comment]
In your case the UI tries to add the resource to a catalog that does not exist (ID: ...7621).
This does not occur for me with bootstrapping enabled and current state on branch "main" of the DSC (commit 9237c12, 14.09.21) and current state of branch "develop" of the UI (commit 750b435, 14.09.21).

**Comment by BastianWel** [comment]
Currently the UI requests all catalogs from DSC and uses the first catalog as default.
After that the default catalog ID is saved in the UI.
If the default catalog in the DSC is deleted afterwards (e.g. via Swagger), then the above error can occur, since the old ID is still used.
Saving the ID in the UI will be removed and the default catalog will be determined each time.

**Comment by BastianWel** [comment]
Fixed in 9b295b9
